src/Core/Framework/Adapter/Cache/Http/CacheResponseSubscriber.php

```diff
@@ -78,6 +97,28 @@ public function setResponseCache(ResponseEvent $event): void
             // To still be able to serve 404 pages fast, we don't load the full context and cache the rendered html on application side
             // as we don't have the full context the state handling is broken as no customer or cart is available, even if the customer is logged in
             // @see \Shopware\Storefront\Framework\Routing\NotFound\NotFoundSubscriber::onError
+            $this->noCache($request, $response, $isStoreApi);
+
+            return;
+        }
+
+        // Normalize attribute value to CacheAttribute instance or null
+        /** @var CacheAttributeType $cacheAttributeValue */
+        $cacheAttributeValue = $request->attributes->get(PlatformRequest::ATTRIBUTE_HTTP_CACHE);
+        $cacheAttribute = CacheAttribute::fromAttributeValue($cacheAttributeValue);
+
+        // In Store API we rely on headers to manage caching, as it more explicit and easier to parse on reverse proxy side.
```

Comment:
we also use headers for storefront now, i thought we would be able to combine that logic so that we don't have two different branches of logic here

Optimaly we would only have some `applyCachePolicy` method and the check whether it is storefront or not should happen in there, but not on the layer above